panther_hardware_interfaces/test/test_panther_imu.cpp

```diff
@@ -0,0 +1,327 @@
+  }
+
+  void CheckSensor() const { PantherImuSensor::CheckSensor(); }
+
+  void CheckStatesSize() const { PantherImuSensor::CheckStatesSize(); }
+
+  void CheckInterfaces() const { PantherImuSensor::CheckInterfaces(); }
+  void ReadObligatoryParams() { PantherImuSensor::ReadObligatoryParams(); }
+
+  void ConfigureMadgwickFilter() { PantherImuSensor::ConfigureMadgwickFilter(); }
+
+  void HandleFirstDataCallback(
+    const geometry_msgs::msg::Vector3 & mag_compensated,
+    const geometry_msgs::msg::Vector3 & lin_acc, const double timestamp_s)
+  {
+    PantherImuSensor::HandleFirstDataCallback(mag_compensated, lin_acc, timestamp_s);
+  }
+};
+
+class TestPantherImuSensor : public testing::Test
+{
+public:
+  TestPantherImuSensor() { imu_sensor_ = std::make_unique<PantherImuSensorWrapper>(); }
+
+  ~TestPantherImuSensor() {}
+
+protected:
+  std::unique_ptr<PantherImuSensorWrapper> imu_sensor_;
+};
+
+TEST_F(TestPantherImuSensor, CheckSensor)
+{
+  hardware_interface::HardwareInfo info;
+  imu_sensor_->SetHardwareInfo(info);
+  EXPECT_THROW({ imu_sensor_->CheckSensor(); }, std::runtime_error);
+
+  hardware_interface::ComponentInfo sensor_info;
+  sensor_info.name = "wrong_imu";
+  info.sensors.push_back(sensor_info);
+  imu_sensor_->SetHardwareInfo(info);
+  EXPECT_THROW({ imu_sensor_->CheckSensor(); }, std::runtime_error);
+
+  info.sensors.front().name = "imu";
+  imu_sensor_->SetHardwareInfo(info);
+  EXPECT_NO_THROW({ imu_sensor_->CheckSensor(); });
+}
+
+TEST_F(TestPantherImuSensor, CheckStatesSize)
+{
+  hardware_interface::HardwareInfo info;
+  info.sensors.push_back({});
+  imu_sensor_->SetHardwareInfo(info);
+
+  EXPECT_THROW({ imu_sensor_->CheckStatesSize(); }, std::runtime_error);
+
+  std::list<std::string> example_interfaces_names = {"state1", "state2", "state3", "state4",
+                                                     "state5", "state6", "state7", "state8",
+                                                     "state9", "state10"};
+  for (const auto & interface_name : example_interfaces_names) {
+    hardware_interface::InterfaceInfo interface_info;
+    interface_info.name = interface_name;
+    info.sensors.front().state_interfaces.push_back(interface_info);
+  }
+
+  imu_sensor_->SetHardwareInfo(info);
+  EXPECT_NO_THROW({ imu_sensor_->CheckStatesSize(); });
+}
+
+TEST_F(TestPantherImuSensor, CheckInterfaces)
+{
+  hardware_interface::HardwareInfo info;
+  info.sensors.push_back({});
+
+  std::list<std::string> example_interfaces_names = {"state1", "state2", "state3", "state4",
+                                                     "state5", "state6", "state7", "state8",
+                                                     "state9", "state10"};
+  for (const auto & interface_name : example_interfaces_names) {
+    hardware_interface::InterfaceInfo interface_info;
+    interface_info.name = interface_name;
+    info.sensors.front().state_interfaces.push_back(interface_info);
+  }
```

Comment:
The same lines of code are present in the CheckStatesSize method. Could we extract the for loop and info initialization into a separate function?

Reply:
Created:
- `hardware_interface::HardwareInfo CreateExampleInterfaces(const hardware_interface::HardwareInfo &info)`
- `hardware_interface::HardwareInfo CreateCorrectInterfaces(const hardware_interface::HardwareInfo &info)`